### DISCUSSION TITLE
Mismatch in Generated Images on iOS 18 vs. Previous Versions

### DISCUSSION THREAD

**Discussion by kyyoussef** [pointfreeco/swift-snapshot-testing#928] [General]
**Describe the bug**

* The `assertSnapshots` method is producing different images for the same view compared to earlier iOS versions. After upgrading my testing devices to iOS 18, my test cases have started to fail.

**To Reproduce**

```swift

struct ContentView: View {
    var body: some View {
        HStack {
            VStack {
                Text("Hello")
                Text("Hello, world!")
                HStack {
                    Text("Hello, world!")
                    Text("Hello, world!")
                }
                HStack {
                    Text("Hello, world!")
                    Text("Hello, world!")
                    Text("Hello, world!")

                }
                Text("Hello, world!")
                Text("Hello, world!")

            }
            .padding()
        }
    }
}

// testing method 
    func testUI() throws {
    
// method is pass either on iOS 18 or 17  (depending on the record on the disk)
        assertSnapshots(of: ContentView(), as: ["": .wait(for: 1,
                                                          on: .image(precision: 1,
                                                                     layout: .fixed(width: 375, height: 667)))])
    }
```

**Expected behavior**

* consistent image generated.

**Environment**
  - swift-snapshot-testing version [e.g. 1.9.0]
  - Xcode Version 15.3
  - Swift [e.g. 5.5]
  - OS: iOS 18 and iOS 17

**Comment by mbrandonw** [comment]
Hi @kyyoussef, this is to be expected. iOS updates can cause small changes in how views are laid out. You really have no choice but to re-record all of your images.

Since this isn't an issue with the library I am going to convert it to a discussion. Please feel free to continue the conversation over there!

**Comment by kyyoussef** [comment]
@mbrandonw Thank you, is there any way to decouple my tests from the os version ?

**Comment by Rion-Kaneshiro** [comment]
@mbrandonw Wouldn't that be a feature request? Especially now with the big changes with iOS 26 and supporting multiple OS-Versions that will be vastly different going forward.

In a client project I also face the situation, that we run the test on iOS 17, 18 and 26.

As of now this is not possible with Snapshot Testing, unfortunately.
